Fix edit mode entity update bug

MoonPadUSer opened the pull request:
as described in issue #64 triggers wouldn't work on entities that you have never touched (literally, once you clicked on them, they were added to a list and they'd start working) ,I fixed this by calling the function that gets called when you edit an entity (`entities::editent(...)`) when the server sends an `N_EDITENT` event

TheAssassin commented on src/game/entities.cpp:
```diff
@@ -1254,12 +1254,18 @@ namespace entities
     }
 
-    void editent(int i)
+    void editent(int i, bool local)
```
Please document the meaning of this new variable. It is not obvious at all.

TheAssassin commented on src/shared/igame.h:
```diff
@@ -4,7 +4,7 @@ namespace entities
     extern int numattrs(int type);
     extern int triggertime(extentity &e);
-    extern void editent(int i);
+    extern void editent(int i, bool local = true);
```
Same here. Variable should be documented.

TheAssassin commented on src/engine/world.cpp:
```diff
@@ -1183,5 +1183,6 @@ void mpeditent(int i, const vec &o, int type, attrvector &attr, bool local)
         e.attrs.add(0, max(entities::numattrs(e.type), min(attr.length(), MAXENTATTRS)) - e.attrs.length());
         loopk(min(attr.length(), e.attrs.length())) e.attrs[k] = attr[k];
         addentity(i);
+        entities::editent(i, false);
```
You need to document this call in-code. Write a comment which explains why you call this function here.

MoonPadUSer commented:
I changed the variable names so they are more descriptive (at least I hope they are) and added one comment. I hope this is enough but I am very unsure about it.

bmorel commented on src/game/entities.cpp:
```diff
@@ -1254,16 +1254,21 @@ namespace entities
+            m_edit(game::gamemode) &&
+            game::player1.state == CS_EDITING)
+        {
+            client::addmsg(N_EDITENT, "ri5iv", i, (int)(e.o.x*DMF), (int)(e.o.y*DMF), (int)(e.o.z*DMF), e.type, e.attrs.length(), e.attrs.length(), e.attrs.getbuf()); // FIXME
```
since #59, size() can be used for vector. Since that line is touched, why not use size() instead of length()?
Also, we're using C++, not C, take the occasion to remove the C-style casts when possible.

bmorel commented on src/game/entities.cpp:
```diff
@@ -1254,12 +1254,18 @@ namespace entities
+            fixentity(i, true);
+            if(m_edit(game::gamemode) && game::player1.state == CS_EDITING)
+            {
+                client::addmsg(N_EDITENT, "ri5iv", i, (int)(e.o.x*DMF), (int)(e.o.y*DMF), (int)(e.o.z*DMF), e.type, e.attrs.length(), e.attrs.length(), e.attrs.getbuf()); // FIXME
```
C-style cast should be avoided. Since #59 we can use STL-compatible methods for vector in most cases.
